doc/applications.md

```diff
@@ -0,0 +1,151 @@
+# Creating applications
+
+RxJS is a great tool to keep your code less error prone. It does that by using pure and stateless functions. But applications are stateful, so how do we bridge the stateless world of RxJS with the stateful world of our applications?
+
+### Vanilla JS
+```js
+const button = document.querySelector('button');
+Rx.Observable.fromEvent(button, 'click')
+  .scan(count => count + 1, 0)
+  .subscribe(count => document.querySelector('#count').innerHTML = count);
+```
+So producing state is within the world of RxJS, but changing the DOM is a side effect which happens at "the end of the line".
+
+### jQuery
+```js
+Rx.Observable.fromEvent($('button'), 'click')
+  .scan(count => count + 1, 0)
+  .subscribe(count => document.querySelector('#count').innerHTML = count);
+```
+
+## State stores
+Applications use state stores to hold state. These are called different things in different frameworks, like store, reducer and model, but at the core they are all just a plain object. What we also need to handle is that multiple observables can update a single state store.
+
+```js
+// ES2015
+const increaseButton = document.querySelector('#increase');
+const increase$ = Rx.Observable.fromEvent(increaseButton, 'click')
+  .map(() => state => {...state, count: state.count + 1});
```

Comment:
Object spread `...` is unfortunately not ES2015, so I'd avoid using that in official docs.